Sources/OktaIdx/Internal/Implementations/Version1/Responses/Responses.swift

```diff
@@ -263,4 +263,62 @@ extension IDXClient.APIVersion1 {
         let refreshToken: String?
         let idToken: String?
     }
+
+    struct Redirect {
+        enum Query {
+            enum Key {
+                static let interactionCode = "interaction_code"
+                static let state = "state"
+                static let error = "error"
+                static let errorDescription = "error_description"
+            }
+            
+            enum Value {
+                static let interactionRequired = "interaction_required"
+            }
+        }
+        
+        let url: URL
+        let scheme: String
+        let path: String
+        
+        let interactionCode: String?
+        let state: String?
+        let error: String?
+        let errorDescription: String?
+        
+        let interactionRequired: Bool
+        
+        init?(url: String) {
+            guard let url = URL(string: url) else {
+                return nil
+            }
+            
+            self.init(url: url)
+        }
+
+        init?(url: URL) {
+            self.url = url
+            
+            guard let urlComponents = URLComponents(string: url.absoluteString), let scheme = urlComponents.scheme else {
```

Comment:
Can you split this across multiple lines, and place the curly brace on its own line?